packages/flutter/lib/src/material/chip.dart

```diff
@@ -1834,7 +1834,7 @@ class _RawChipState extends State<RawChip> with TickerProviderStateMixin<RawChip
 
     final ThemeData theme = Theme.of(context)!;
     final ChipThemeData chipTheme = ChipTheme.of(context);
-    final TextDirection? textDirection = Directionality.of(context);
+    final TextDirection? textDirection = Directionality.maybeOf(context);
```

Comment:
Here and below: should this really be nullable? I understand it was, but it might help to have a comment here about why we're allowing this to be null instead of using .of.

Reply:
Ultimately, they are passed to EdgeInsetsGeometry.resolve, which is fine with a null text direction as long as the EdgeInsetsGeometry is not text direction aware. Similar with the other cases.

We actually have tests for these. If you make it non-null here, they will fail.